src/main/java/org/broadinstitute/hellbender/tools/spark/sv/discovery/prototype/SuspectedTransLocDetector.java

```diff
@@ -0,0 +1,71 @@
+import org.broadinstitute.hellbender.utils.Utils;
+import scala.Tuple2;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+final class SuspectedTransLocDetector implements VariantDetectorFromLocalAssemblyContigAlignments {
+
+    @SuppressWarnings("unchecked")
+    private static final List<String> EMPTY_INSERTION_MAPPINGS = Collections.EMPTY_LIST;
+
+    @Override
+    public void inferSvAndWriteVCF(final JavaRDD<AlignedContig> localAssemblyContigs, final String vcfOutputFileName,
+                                   final Broadcast<ReferenceMultiSource> broadcastReference,
+                                   final Broadcast<SAMSequenceDictionary> referenceDictionaryBroadcast,
+                                   final Logger toolLogger) {
+        localAssemblyContigs.cache();
+        toolLogger.info(localAssemblyContigs.count() + " chimeras indicating strand-switch-less breakpoints");
+
+        final JavaPairRDD<ChimericAlignment, byte[]> chimeraAndSequence =
+                localAssemblyContigs
+                        .filter(tig ->
+                                SimpleStrandSwitchVariantDetector.splitPairStrongEnoughEvidenceForCA(tig.alignmentIntervals.get(0),
+                                        tig.alignmentIntervals.get(1), SimpleStrandSwitchVariantDetector.MORE_RELAXED_ALIGNMENT_MIN_MQ,
+                                        0))
+                        .mapToPair(tig -> convertAlignmentIntervalsToChimericAlignment(tig, referenceDictionaryBroadcast.getValue())).cache();
+
+        final JavaRDD<VariantContext> annotatedBNDs =
+                chimeraAndSequence
+                        .mapToPair(pair -> new Tuple2<>(new NovelAdjacencyReferenceLocations(pair._1, pair._2, referenceDictionaryBroadcast.getValue()), pair._1))
+                        .groupByKey()
+                        .mapToPair(noveltyAndEvidence -> inferBNDType(noveltyAndEvidence, broadcastReference.getValue(), referenceDictionaryBroadcast.getValue()))
+                        .flatMap(noveltyTypeAndEvidence ->
+                                AnnotatedVariantProducer
+                                        .produceAnnotatedBNDmatesVcFromNovelAdjacency(noveltyTypeAndEvidence._1,
+                                                noveltyTypeAndEvidence._2._1, noveltyTypeAndEvidence._2._2, broadcastReference).iterator());
+
+        SVVCFWriter.writeVCF(annotatedBNDs.collect(), vcfOutputFileName.replace(".vcf", "_transBND.vcf"),
+                            referenceDictionaryBroadcast.getValue(), toolLogger);
+    }
+
+    private static Tuple2<ChimericAlignment, byte[]> convertAlignmentIntervalsToChimericAlignment(final AlignedContig contig,
+                                                                                                  final SAMSequenceDictionary referenceDictionary) {
+        // TODO: 9/9/17 this default empty insertion mapping treatment is temporary and should be fixed later
```

Comment:
What needs to happen to fix this?

Reply:
removed the comment.
What I really was thinking about is that the alignment scoring and filtering step might decide an alignment brings in noise and filter it away,  think about the following scenario: 
a configuration with 3 alignments, with the middle one overlaps with other two, and has a mediocre MQ, this currently is treated as noise and filtered away in the alternative code path, but the master code path would note down it as inserted sequence mapping and skip it. 
The above scenario is definitely one place where the filtering step should be improved upon (#3473 and #3647 )